models/secret/secret.go

```diff
@@ -21,8 +21,8 @@ import (
 // Secret represents a secret
 type Secret struct {
 	ID          int64
-	OwnerID     int64              `xorm:"INDEX UNIQUE(owner_repo_name) NOT NULL"`
-	RepoID      int64              `xorm:"INDEX UNIQUE(owner_repo_name) NOT NULL DEFAULT 0"`
+	OwnerID     *int64             `xorm:"INDEX UNIQUE(owner_repo_name)"`
```

Comment:
We can define if ownerID == 0 and repoID == 0, then it's a global secret. What's the benefit to use a pointer of int64?

Reply:
I updated the commit :)

Reply:
Made a separate PR for it here : #27829